cmd/rook/ceph/config.go

```diff
@@ -18,81 +18,74 @@ package ceph
 
 import (
 	"fmt"
+	"io/ioutil"
+	"os"
 
 	"github.com/rook/rook/cmd/rook/rook"
 	cephconfig "github.com/rook/rook/pkg/daemon/ceph/config"
-	"github.com/rook/rook/pkg/operator/ceph/cluster/mon"
 	"github.com/rook/rook/pkg/util"
-	"github.com/rook/rook/pkg/util/flags"
 	"github.com/spf13/cobra"
 )
 
-const (
-	configKeyringTemplate = `
-[%s]
-key = %s
-`
-)
-
 var configCmd = &cobra.Command{
 	Use:   "config-init",
-	Short: "Generates basic ceph config",
+	Short: "Generates basic Ceph config",
+	Long: `Generate the most basic Ceph config for connecting non-Ceph daemons to a Ceph
+cluster (e.g., nfs-ganesha). Effectively what this means is that it generates
+'/etc/ceph/ceph.conf' with 'mon_host' populated and a keyring path (given via
+commandline flag) associated with the user given via commandline flag.
+'mon_host' is determined by the 'ROOK_CEPH_MON_HOST' env var present in other
+Ceph daemon pods, and the keyring is expected to be mounted into the container
+with a Kubernetes pod volume+mount.`,
 }
 
 var (
-	configKeyring  string
-	configUsername string
+	keyring  string
+	username string
 )
 
 func init() {
-	configCmd.Flags().StringVar(&configKeyring, "keyring", "", "the daemon keyring")
-	configCmd.Flags().StringVar(&configUsername, "username", "", "the daemon username")
-	addCephFlags(configCmd)
+	configCmd.Flags().StringVar(&keyring, "keyring", "", "path to the keyring file")
+	configCmd.MarkFlagRequired("keyring")
 
-	flags.SetFlagsFromEnv(configCmd.Flags(), rook.RookEnvVarPrefix)
+	configCmd.Flags().StringVar(&username, "username", "", "the daemon username")
+	configCmd.MarkFlagRequired("username")
 
 	configCmd.RunE = initConfig
 }
 
 func initConfig(cmd *cobra.Command, args []string) error {
-	required := []string{
-		"username", "keyring", "mon-endpoints"}
-	if err := flags.VerifyRequiredFlags(configCmd, required); err != nil {
```

Comment:
> LGTM, just a question about the testing. The biggest change to the approach I want to make sure is tested is with any dbus interaction since it's running in a sidecar instead of the same container. @jtlayton Your testing would have exercised the dbus, right?

No, I didn't actually test dbus. The project I was working on before didn't use it.

Reply:
I spun up another cluster and tested that I could reach dbus. I didn't test anything specific, but it seems like it still works. I exec'ed into the ganesha container and did this:
```
# dbus-send --system             --dest=org.freedesktop.DBus   --type=method_call            --print-reply                 /org/freedesktop/DBus         org.freedesktop.DBus.ListNames
method return time=1563892479.192571 sender=org.freedesktop.DBus -> destination=:1.2 serial=3 reply_serial=2
   array [
      string "org.freedesktop.DBus"
      string ":1.0"
      string "org.ganesha.nfsd"
      string ":1.2"
   ]
```
It's not clear to me how people will generally use dbus commands under rook, but this method seems to work ok.